pkg/monitor/monitor/util.go

```diff
@@ -638,6 +638,21 @@ func getMonitorService(monitor *v1alpha1.TidbMonitor) []*core.Service {
 	var services []*core.Service
 	monitorLabel := label.New().Instance(monitor.Name).Monitor()
 	labels := label.NewMonitor().Instance(monitor.Name).Monitor()
+
+	reloaderPortName := "tcp-reloader"
+	prometheusPortName := "http-prometheus"
+	grafanaPortName := "http-grafana"
+
+	if monitor.Spec.Reloader.Service.PortName != nil {
```

Comment:
what if `PortName` is non-nil but empty, should we use the default name in that case?
cc @aylei do you recommend to use `*string` or `string`?

Reply:
> what if `PortName` is non-nil but empty, should we use the default name in that case?
> cc @aylei do you recommend to use `*string` or `string`?

I think operator don't need to consider this case, if monitor service synced failed because of setting wrong by users, users should be responsible for it.

Reply:
It is about our purpose. For kubernetes, port name is optional and could be empty, so the current code is okay. If we think empty is not a valid value, then we could consider empty as `unset` and use `string`, otherwise, just keep `*string`